### DISCUSSION TITLE
What software do you guys use for streaming videos to the server?

### DISCUSSION THREAD

**Discussion by Wazbat** [zorchenhimer/MovieNight#164] [Q&A]
I've been using OBS myself with the "VLC Video Source" to stream videos, however it's a bit of a pain.

Are there any better/easier alternatives?

**Comment by zorglube** [comment]
You can use https://ffmpeg.org/ with a "kind of" tricky command line.  
You can certainly directly use VLC.

**Comment by CptPie** [comment]
Technically any tool that outputs an rtmp stream should work fine. If you have issues with the VLC source, you might as well capture any videoplayer as a normal windowcapture and stream normally to the MovieNight server.

**Reply by Wazbat** [reply]
Yeah that's fair. I'm just curious what options others use.

Might even be possible to stream something like Kodi, then use the [Kodi event server](https://kodi.wiki/view/JSON-RPC_API/v10#Input) to control it, maybe even using chat commands that forward the controls

**Reply by zorglube** [reply]
> Yeah that's fair. I'm just curious what options others use.
> 
> Might even be possible to stream something like Kodi, then use the [Kodi event server](https://kodi.wiki/view/JSON-RPC_API/v10#Input) to control it, maybe even using chat commands that forward the controls

What do you mean by "control", witch part would you like to control ?

**Reply by Wazbat** [reply]
The user interface. I was thinking of running Kodi and streaming the interface. The interface has options for selecting media, controlling subtitles, etc. Trusted users could control said interface remotely

Another example is viewing this stream inside of multiplayer games like NeosVR. You can code buttons to fire off web requests, which in turn would in theory allow you to control this interface remotely, while viewing the stream in game. I sort of tried to do this and  it worked pretty well. You just set the player in game to view the /live endpoints and make sure that it's http and not https, and it seems to work fine. Would make for a fun interactive experience

**Reply by zorglube** [reply]
Ok, so you mean controlling the stream-source through the MoviNight chat. AM I right ?

**Reply by Wazbat** [reply]
For example. That's one thing that's possible by streaming an application like Kodi.

Just curious to hear about how others are using MovieNight

**Reply by zorglube** [reply]
My side I use it only as broadcast relay to watch movie with friends.
